src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -308,7 +308,10 @@ void Transformations::PreLpt(const std::vector<ov::element::Type>& defaultPrecis
     ov::pass::Manager decompression_handling_manager;
     decompression_handling_manager.set_per_pass_validation(false);
     CPU_REGISTER_PASS_COMMON(decompression_handling_manager, ov::pass::InitNodeInfo);
-    CPU_REGISTER_PASS_COMMON(decompression_handling_manager, ov::pass::ConvertGatherToGatherCompressed);
+    const bool useLpt = !defaultPrecisions.empty();
+    if (!useLpt) {
```

Comment:
Another thing we should care about is how to preserve performance on hybrid models (both quantized and compressed, like in https://jira.devtools.intel.com/browse/CVS-124535).
The idea is the following: lets disable `ConvertGatherToGatherCompressed` pass only for emdedding precision supported by LPT (assuming dequantizationwill be handled by LPT properly). To do we should register pass unconditionally. but update the callback on line 328: if GatherCompressed input has U8 or I8 precison and useLpt is true, then the callback should return true.

Reply:
GatherCompressed only process compressed weights(u8/i8/u4/i4), you mean we just process u8/i8 in this case?
Will update firstly.

Reply:
Yeah, the point is CPU quantization pipeline (handled by LPT) supports only u8/i8. So for i4/u4 we still need to enable ConvertGatherToGatherCompressed optimization even if useLPT = true